frontend_csv/src/scanner/mod.rs

```diff
@@ -118,6 +119,10 @@ mod test {
         result.reverse();
         assert!(result.len() > 0);
         let expected = vec![
+            token::CSVToken {
+                ty: token::CSVTokenType::EOF,
+                val: "".to_string(),
+            },
```

Comment:
the EOF should be at the end?

Reply:
Yes for easy processing, with EOF at the end you don't need to check whether or not buffer is empty at end of iteration